plugins/inputs/win_wmi/README.md

```diff
@@ -0,0 +1,229 @@
+# Windows Management Interface Input Plugin
+
+This document presents the input plugin to read WMI classes on Windows
+operating systems. With the win_wmi plugin, an administrator is enabled to
+capture and filter virtually any configuration or metric value exposed through
+the Windows Management Instrumentation service.
+
+If a WMI class property's value is a string, then the string is included with
+the metric as a tag. If a WMI class property's value is an integer, then the
+integer is used as a metric field.
+
+If telegraf is configured with a logfile and the plugin's configuration
+contains an invalid namespace, class, or property, an error is logged.
+
+## Basics
+
+The examples contained in this file have been utilized and exercised in Windows
+environments. There are many other useful classes to monitor if you know what
+to look for.
+
+### Namespace
+
+A string representing the WMI namespace to be queried. For example,
+`root\\cimv2`.
+
+### ClassName
+
+A string representing the WMI class to be queried. For example,
+`Win32_Processor`.
+
+### Properties
+
+An array of strings representing the properties of the WMI class to be queried.
+By default, the `Name` property is included in the query. However, some classes
+do not contain a `Name` property, in which cases the `ExcludeNameKey`
+configuration should be utilized.
+
+### ExcludeNameKey
+
+By default, a WMI class's `Name` property is included as a tag value in order
+for each metric to have a unique identifier. However, some WMI classes do not
+have a `Name` property. In such cases, ExcludeNameKey should be set to `True`.
+
+## Configuration
+
+```toml @sample.conf
+  ## By default, this plugin returns no results.
+  ## Uncomment the example below or write your own as you see fit.
+  ## The "Name" property of a WMI class is automatically included unless
+  ## excludenamekey is true.
+  ## If the WMI property's value is a string, then it is used as a tag.
+  ## If the WMI property's value is a type of int, then it is used as a field.
+  ## [[inputs.win_wmi]]
+  ##   namespace = "root\\cimv2"
+  ##   classname = "Win32_Volume"
+  ##   properties = ["Capacity", "FreeSpace"]
+  ##   filter = 'NOT Name LIKE "\\\\?\\%"'
+  ##   excludenamekey = false
+  ##   name_prefix = "win_wmi_"
+```

```

Comment:
This style of configuration is a little unusual for Telegraf. Usually a plugin instance can gather multiple things. Currently this plugin requires a separate plugin instance for each WMI query.